### PR TITLE
refactor(auth): decouple sign-in UI from BibleReaderView and add global sign-in config

### PR DESCRIPTION
**Pull request by bryanmontz**
## Summary

- **Moves sign-in configuration to `YouVersionPlatformConfiguration`** — `appName`, `signInPromptMessage`, and a new `isSignInEnabled` toggle are now set once at app startup via `configure()` instead of being passed through every `BibleReaderView` initializer
- **Extracts `SignInView` into `YouVersionPlatformUI`** as a standalone public view with no `BibleReaderViewModel` dependency — it accepts action closures (`onSignIn`, `onDismiss`) and theming colors via init, making it reusable outside the Bible reader context
- **Adds `isSignInEnabled` kill switch** that suppresses all sign-in UI: verse-tap prompts, version-download auth checks, and the header menu sign-in/sign-out buttons
- **Deprecates `BibleReaderView(appName:signInMessage:)`** with a backward-compatible shim that forwards to `configureSignIn()`, so existing call sites get a warning but continue to work
- **Moves `BigButtonStyle` to `YouVersionPlatformUI`** and makes it public so both modules can use it

## Migration guide

**Before:**
```swift
BibleReaderView(
    appName: "My App",
    signInMessage: "Sign in to see highlights."
)
```

**After:**
```swift
// In app startup (e.g. App.init):
YouVersionPlatformConfiguration.configure(
    appKey: "...",
    appName: "My App",
    isSignInEnabled: true,
    signInPromptMessage: "Sign in to see highlights."
)

// In views:
BibleReaderView()
```

To disable all sign-in UI, set `isSignInEnabled: false` and provide an `onVerseTap` closure.

## Test plan

- [ ] Verify `BibleReaderView()` (new init) works with sign-in configured globally
- [ ] Verify deprecated `BibleReaderView(appName:signInMessage:)` still compiles with a deprecation warning
- [ ] Verify `isSignInEnabled = false` suppresses: verse-tap sign-in sheet, header menu sign-in button, version-download auth prompt
- [ ] Verify `SignInView` renders correctly when presented standalone (outside `BibleReaderView`)
- [ ] Verify version download still triggers sign-in when `isSignInEnabled = true` and user is unauthenticated

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR nicely decouples sign-in configuration from `BibleReaderView` by moving `appName`, `signInPromptMessage`, and the new `isSignInEnabled` kill-switch into `YouVersionPlatformConfiguration`, and extracts the sign-in UI into a standalone, reusable `SignInWithYouVersionView`. The backward-compatible deprecated shim and the updated `configureSignIn` (now correctly `@MainActor` with `isSignInEnabled = true`) show good attention to the migration concerns raised earlier.

Two small polish items in the new `SignInWithYouVersionView` worth a look before shipping: (1) `appName` silently falls back to `\"\"`, which produces awkward sign-in paragraph text when the caller hasn't configured it, and (2) the logo loses the `.environment(\\.colorScheme, …)` override the old view applied, so it may not match the reader's configured theme.

<h3>Confidence Score: 5/5</h3>

Safe to merge — all remaining findings are P2 polish items that don't block functionality.

The previously-flagged P1 issues (missing @MainActor on configureSignIn, deprecated shim not restoring isSignInEnabled) are fully resolved. The three remaining findings are all P2: an empty-appName edge case, a lost colorScheme override on the logo, and an overly-broad test teardown. None affect the primary sign-in flow or data integrity.

Sources/YouVersionPlatformUI/Views/SignInWithYouVersionView.swift — two minor UI polish items worth addressing before wider adoption of the standalone view.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| Sources/YouVersionPlatformCore/YouVersionPlatformConfiguration.swift | Adds appName, signInPromptMessage, and isSignInEnabled to the global config; both configure and the new configureSignIn are correctly @MainActor. Minor: configure(appKey: nil) now erases the key rather than no-oping. |
| Sources/YouVersionPlatformReader/BibleReaderView.swift | Removes appName/appSignInMessage state; deprecated init forwards to configureSignIn; new init has a helpful assert. The signInView loses the colorScheme environment override for the logo. |
| Sources/YouVersionPlatformUI/Views/SignInWithYouVersionView.swift | New reusable public sign-in view with clean closure-based API and good theming support. Two concerns: appName silently falls back to empty string, and the logo loses its explicit colorScheme override. |
| Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift | Updated to use configure() since appKey is now private(set). Teardown via configure(appKey: originalAppKey) inadvertently resets appName, isSignInEnabled, and signInPromptMessage beyond what the tests intend to restore. |
| Sources/YouVersionPlatformReader/Components/BibleReaderDrawer.swift | Gating highlight color buttons behind YouVersionAPI.isSignedIn is a sensible UX improvement preventing unauthenticated users from interacting with highlight actions. |
| Sources/YouVersionPlatformReader/Components/BibleReaderHeaderMenuView.swift | Sign-in/sign-out menu items correctly gated behind isSignInEnabled; clean, straightforward change. |
| Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift | Verse-tap flow correctly checks isSignedIn then isSignInEnabled; silent no-op for unauthenticated + sign-in-disabled is intentional per design. |
| Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+MyVersions.swift | Adds isSignInEnabled guard before triggering sign-in on version download; assertionFailure for the disabled-but-unauthenticated path is appropriate per the author's stated design intent. |
| Sources/YouVersionPlatformUI/Views/YouVersionBigButtonStyle.swift | Promoted from internal BigButtonStyle to public YouVersionBigButtonStyle in YouVersionPlatformUI; clean rename with public memberwise init added. |
| Sources/YouVersionPlatformReader/Sheets/BibleReaderSignInView.swift | Deleted and superseded by the new standalone SignInWithYouVersionView; no concerns. |
| Sources/YouVersionPlatformReader/Sheets/BibleReaderVersionInfoView.swift | Straightforward rename of BigButtonStyle to YouVersionBigButtonStyle; no functional changes. |
| Sources/YouVersionPlatformReader/Sheets/BibleVersionDownloadView.swift | Renamed BigButtonStyle to YouVersionBigButtonStyle and cleaned up trailing whitespace; no functional changes. |
| AGENTS.md | Added naming convention rule requiring public-facing types to include YouVersion in their name; consistent with the new types introduced in this PR. |
| README.md | Updated SDK documentation to reflect the new configure() API, migration guide, and SignInWithYouVersionView usage. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[User taps verse] --> B{onVerseTap closure set?}
    B -- Yes --> C[Call onVerseTap closure\nhost app handles]
    B -- No --> D{YouVersionAPI.isSignedIn?}
    D -- Yes --> E[Toggle verse selection\nShow actions drawer]
    D -- No --> F{isSignInEnabled?}
    F -- Yes --> G[Show SignInWithYouVersionView sheet]
    F -- No --> H[Silent no-op\nassertionFailure in debug]
    G --> I{User taps Sign In?}
    I -- Yes --> J[viewModel.signIn / Dismiss sheet]
    I -- No --> K[Dismiss sheet]
    style C fill:#d4edda
    style E fill:#d4edda
    style G fill:#fff3cd
    style H fill:#f8d7da
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: Sources/YouVersionPlatformUI/Views/SignInWithYouVersionView.swift
Line: 47-49

Comment:
**Empty `appName` renders awkwardly in sign-in paragraph**

When `YouVersionPlatformConfiguration.appName` is `nil` (e.g., `configure(appKey:)` called without `appName`), the fallback `""` causes `String(format: localized("signIn.paragraph"), "")` to produce something like "Sign in with YouVersion in ." — a trailing empty token. Consider guarding on the empty state:

```suggestion
    private var appName: String {
        YouVersionPlatformConfiguration.appName ?? Bundle.main.displayName ?? ""
    }
```

Or conditionally hide that `Text` line when `appName` is nil, the same way `signInMessage` is handled above it.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: Sources/YouVersionPlatformUI/Views/SignInWithYouVersionView.swift
Line: 61-64

Comment:
**Logo `colorScheme` override dropped from refactored view**

The old `BibleReaderSignInView` applied `.environment(\.colorScheme, viewModel.colorTheme?.colorScheme == .dark ? .dark : .light)` to this logo image so it would match the reader's configured theme (which can differ from the system theme). The new view has no equivalent override, so on a device in Light Mode using a dark-themed reader, the logo variant may render incorrectly.

The theme's effective color scheme could be threaded through as an optional init parameter — or `BibleReaderView.signInView` could wrap the sheet with `.environment(\.colorScheme, ...)` before presenting it.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift
Line: 121

Comment:
**Teardown has broader side-effects than intended**

The cleanup `configure()` call correctly restores the key, but also resets `appName`, `isSignInEnabled`, and `signInPromptMessage` to their defaults — properties the test never captured. The same pattern repeats at line 144. If a suite-level fixture sets any of those fields before these tests run, this teardown would silently corrupt that state.

Consider capturing all configuration fields at test start and restoring them fully at teardown, or adding a dedicated internal reset helper for use in tests.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (3): Last reviewed commit: ["changes from greptile feedback"](https://github.com/youversion/platform-sdk-swift/commit/63cc8fd4080d53f307472d693e907c1436023745) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28283624)</sub>

### CODE REVIEW

File: Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+MyVersions.swift [hunk 1]
```diff
@@ -71,8 +71,10 @@ extension BibleReaderViewModel {
         Task {
             if await YouVersionAPI.hasValidToken() {
                 finalDownloadButtonTapped(version: version)
-            } else {
+            } else if YouVersionPlatformConfiguration.isSignInEnabled {
                 startSignInFlow = true
+            } else {
+                assertionFailure("YouVersion sign-in must be enabled to download Bible versions.")
             }
         }
```
[the remark this answers is not in the document]
Comment (bryanmontz) [reply Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+MyVersions.swift:79]: This is by design — Bible version downloads require sign-in because the publisher contracts require the user's email address. The `assertionFailure` correctly flags a misconfiguration during development.

File: Sources/YouVersionPlatformCore/YouVersionPlatformConfiguration.swift [hunk 2]
```diff
@@ -34,6 +54,11 @@ public struct YouVersionPlatformConfiguration {
             Self.installId = newId
         }
     }
+    
+    public static func configureSignIn(appName: String, signInPromptMessage: String? = nil) {
+        Self.appName = appName
+        Self.signInPromptMessage = signInPromptMessage
+    }
```
[the remark this answers is not in the document]
Comment (bryanmontz) [reply Sources/YouVersionPlatformCore/YouVersionPlatformConfiguration.swift:61]: Good catch — added `@MainActor` and also set `isSignInEnabled = true` per the other comment. Fixed in the next push.

[the remark this answers is not in the document]
Comment (bryanmontz) [reply Sources/YouVersionPlatformCore/YouVersionPlatformConfiguration.swift:61]: Agreed — `configureSignIn` now sets `isSignInEnabled = true` to preserve the old API's contract. Fixed alongside #2.

File: Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift [hunk 3]
```diff
@@ -92,19 +92,20 @@ extension BibleReaderViewModel {
...
+                showingVerseActionsDrawer = !self.selectedVerses.isEmpty
+            }
+        } else if YouVersionPlatformConfiguration.isSignInEnabled {
+            showingSignInSheet = true
         } else {
-            selectedVerses.insert(reference)
-        }
-        withAnimation(.interpolatingSpring(stiffness: 300, damping: 25)) {
-            showingVerseActionsDrawer = !self.selectedVerses.isEmpty
+            assertionFailure("onVerseTap must be provided OR YouVersion sign-in must be enabled")
```
Comment (davidfedor) [inline Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift]: Can we have this assertionFailure logic in init(), such that this error condition is detected on startup instead of waiting until a user tap (hopefully) happens?

File: Sources/YouVersionPlatformUI/Views/BigButtonStyle.swift [hunk 4]
```diff
@@ -1,12 +1,19 @@
 import SwiftUI
 
-struct BigButtonStyle: ButtonStyle {
-    let strokeColor: Color
-    let backgroundColor: Color
-    let foregroundColor: Color
-    let strokeWidth: CGFloat = 1.5
+public struct BigButtonStyle: ButtonStyle {
```
Comment (davidfedor) [inline Sources/YouVersionPlatformUI/Views/BigButtonStyle.swift]: Since this is becoming public, let's call this YouVersionBigButtonStyle

File: Sources/YouVersionPlatformUI/Views/SignInView.swift [hunk 5]
```diff
@@ -0,0 +1,108 @@
+import SwiftUI
+import YouVersionPlatformCore
+
+/// A reusable sign-in prompt view. Reads ``YouVersionPlatformConfiguration/appName``
+/// and ``YouVersionPlatformConfiguration/signInPromptMessage`` from the global configuration.
+///
+/// The caller provides action closures and theming colors so the view has
+/// no dependency on any specific view model.
+public struct SignInView: View {
```
Comment (davidfedor) [inline Sources/YouVersionPlatformUI/Views/SignInView.swift]: Since this is becoming public, let's call it SignInWithYouVersionView to mirror SignInWithYouVersionButton

File: Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift (hunk 3 above)
Comment (davidfedor) [reply Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift]: Furthermore we will want people to be able to select multiple verses to do Copy or Send operations, even if they're not signed in. You don't need to make changes to the verse action sheet - we will do that as we finish up the highlights UX

File: README.md [hunk 6]
```diff
@@ -173,24 +224,32 @@ In the header of your SwiftUI view, store a strong reference to the `ContextProv
...
         }
     }
 }
 ```
 
-> **Note**: The SDK stores the access token locally, and persists it across app launches. 
+You can also present the SDK's `SignInView` directly in your own sheet or navigation flow:
+
+```swift
+SignInView(
```
Comment (davidfedor) [inline README.md]: adjust naming here too

File: Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift [hunk 7]
```diff
@@ -80,8 +80,8 @@ import Testing
 
     @Test func performRefreshSuccessReturnsNewTokens() async throws {
         let originalAppKey = YouVersionPlatformConfiguration.appKey
-        YouVersionPlatformConfiguration.appKey = "test-app"
-        defer { YouVersionPlatformConfiguration.appKey = originalAppKey }
+        await YouVersionPlatformConfiguration.configure(appKey: "test-app")
```
Comment (davidfedor) [inline Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift:83]: why the "await" here?

File: Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift [hunk 8]
```diff
@@ -122,8 +122,8 @@ import Testing
 
     @Test func performRefreshNon200Throws() async {
         let originalAppKey = YouVersionPlatformConfiguration.appKey
-        YouVersionPlatformConfiguration.appKey = "test-app"
-        defer { YouVersionPlatformConfiguration.appKey = originalAppKey }
+        await YouVersionPlatformConfiguration.configure(appKey: "test-app")
```
Comment (davidfedor) [inline Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift:126]: why the "await" here?

**Review by davidfedor — CHANGES_REQUESTED** [review]
Big picture: looks good, thanks for contributing these changes for flexibility. There's nothing like the feedback from the perspective of a real user!
 
Greptile has a concern about the test cleanup pattern, please address.

File: Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift (hunk 3 above)
Comment (davidfedor) [reply Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift]: I think this is accomplishable by just adding `if YouVersionAPI.isSignedIn {` around the use of `highlightColorButtons` in BibleReaderDrawer, BTW, if you want to give that a try.

File: Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift (hunk 7 above)
Comment (bryanmontz) [reply Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift:83]: Because `YouVersionPlatformConfiguration.configure(...)` is marked `@MainActor` and the tests are not, and `await` is required to move between actors. I will improve this in a follow-up PR.

File: Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift (hunk 8 above)
Comment (bryanmontz) [reply Tests/YouVersionPlatformCoreTests/UsersAuthHelpersTests.swift:126]: Because `YouVersionPlatformConfiguration.configure(...)` is marked `@MainActor` and the tests are not, and `await` is required to move between actors. I will improve this in a follow-up PR.

File: Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift (hunk 3 above)
Comment (bryanmontz) [reply Sources/YouVersionPlatformReader/ViewModels/BibleReaderViewModel+Navigation.swift]: Moved the `assert` and wrapped `highlightColorButtons` with `isSignedIn`.

File: Sources/YouVersionPlatformUI/Views/SignInView.swift (hunk 5 above)
Comment (bryanmontz) [reply Sources/YouVersionPlatformUI/Views/SignInView.swift]: Renamed and added a comment about this naming standard in AGENTS.md.

File: README.md (hunk 6 above)
Comment (bryanmontz) [reply README.md]: Renamed and added a comment about this naming standard in AGENTS.md.

File: Sources/YouVersionPlatformUI/Views/BigButtonStyle.swift (hunk 4 above)
Comment (bryanmontz) [reply Sources/YouVersionPlatformUI/Views/BigButtonStyle.swift]: Renamed and added a comment about this naming standard in AGENTS.md.

**Review by davidfedor — APPROVED** [review]
Thanks!
